gap/SnAnUnknownDegree.gi

```diff
@@ -0,0 +1,713 @@
+    # two elements of H2.
+    H2 := [c, cg, ~[2] ^ cg3, ~[3] ^ cg3, ~[4] ^ cg4];
+    if not commutesWithAtMostOne(ri, x1, H2) then return false; fi;
+    if not commutesWithAtMostOne(ri, x2, H2) then return false; fi;
+    if not commutesWithAtMostOne(ri, x3, H2) then return false; fi;
+    return true;
+end);
+
+# ri : recog info record with group G
+# g : element of G,
+#     should be a k-cycle matching c
+# c : element of G,
+#     should be a 3-cycle
+# r : element of G,
+#     should have at least one moved point in common with g and should fix at
+#     least two moved points of g
+# k : integer,
+#     should be length of cycle g
+#
+# Returns fail or a conjugate of r.
+#
+# W.l.o.g. let g = (1, ..., k) and c = (1, 2, 3).
+# If the input is as assumed, then the algorithm returns a conjugate r ^ x such
+# that r ^ x fixes the points 1, 2 but not the point 3.
+BindGlobal("AdjustCycle",
+function(ri, g, c, r, k)
+    local
+        # list of 4 booleans, is point j fixed point
+        F4,
+        # smallest fixed point
+        f1,
+        # second smallest fixed point
+        f2,
+        # smallest non-fixed point
+        m,
+        # bool, false if |F| < 2 or |F| = k
+        success,
+        # integer, loop variable over [1 .. k]
+        j,
+        # element of G, loop variable
+        t,
+        # conjugating element
+        x;
+    # According to the paper we have:
+    # F := { 1 \leq j \leq k | IsFixedPoint(g, c ^ (g ^ (j - 3)), r) = true }
+    # f1 := smallest number in F
+    # f2 := second smallest number in F
+    # m := smallest number *not* in F
+    # We do not store F explicitely. Instead we store the intersection of F and
+    # {1, 2, 3, 4} in the variable F4.
+    F4 := [false, false, false, false];
+    f1 := fail;
+    f2 := fail;
+    m := fail;
+    t := c ^ (g ^ -3);
+    for j in [1 .. k] do
+        # invariant: t = c ^ (g ^ (j - 3))
+        t := t ^ g;
+        if IsFixedPoint(ri, g, t, r) then
+            if j <= 4 then
+                F4[j] := true;
+            fi;
+            if f1 = fail then
+                f1 := j;
+            elif f2 = fail then
+                f2 := j;
+            fi;
+        elif m = fail then
+            m := j;
+        fi;
+        # f1 and f2 not being fail is equivalent to |F| >= 2
+        # m not being fail is equivalent to |F| < k
+        success := f1 <> fail and f2 <> fail and m <> fail;
+        if success then
+            if j >= 4 then
+                break;
+            # 2. Case, we do not need to compute F4[4]
+            elif f1 = 1 and f2 = 2 and m = 3 then
+                return r;
+            fi;
+        fi;
+    od;
+    if not success then
+        return fail;
+    fi;
+    # case distinction on F as in the table of Algorithm 4.20
+    # via a decision tree
+    if F4[1] then
+        if F4[2] then
+            # We are in the 1. Case, since the 2. Case is handled during the
```

Comment:
Change `Case` -> `case` everywhere. Also the `1. case` reads really German. I'd write `first case`, or (more likely) `case 1`.

Reply:
done